src/main/java/org/zalando/baigan/service/RepositoryFactory.java

```diff
@@ -0,0 +1,44 @@
+package org.zalando.baigan.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.zalando.baigan.service.aws.S3ConfigurationRepositoryBuilder;
+
+import java.util.List;
+
+@Component
+public class RepositoryFactory {
+
+    private final ConfigurationParser configurationParser;
+
+    @Autowired
+    public RepositoryFactory(final ConfigurationParser configurationParser) {
+        this.configurationParser = configurationParser;
+    }
+
+    public S3ConfigurationRepositoryBuilder s3ConfigurationRepository() {
+        return new S3ConfigurationRepositoryBuilder(configurationParser);
+    }
+
+    public FileSystemConfigurationRepositoryBuilder fileSystemConfigurationRepository() {
+        return new FileSystemConfigurationRepositoryBuilder(configurationParser);
+    }
+
+    public EtcdConfigurationRepository etcdConfigurationRepository(String etcdUrl) {
+        return new EtcdConfigurationRepository(etcdUrl, configurationParser);
+    }
+
+    public ChainedConfigurationRepository chainedConfigurationRepository(List<ConfigurationRepository> configurationRepositories) {
+        if (configurationRepositories.size() == 2) {
+            return new ChainedConfigurationRepository(configurationRepositories.get(0), configurationRepositories.get(1));
+        } else if (configurationRepositories.size() > 2) {
```

Comment:
Shall we gift `ChainedConfigurationRepository` a new constructor that handles our arbitrary sized list a little bit more convenient?

Reply:
Done in https://github.com/zalando-stups/baigan-config/pull/84